.github/workflows/main.yml

```diff
@@ -107,3 +107,39 @@ jobs:
         echo "debian" $(cat /etc/debian_version)
         /usr/cyrus/libexec/master -V
         /usr/cyrus/sbin/cyr_buildinfo
+  valgrind-make-check:
+    runs-on: ubuntu-latest
+    container:
+        image: ghcr.io/cyrusimap/cyrus-docker:bookworm
+        options: --init
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: setup git safe directory
+      shell: bash
+      run: git config --global --add safe.directory /__w/cyrus-imapd/cyrus-imapd
+    - name: fetch upstream release tags
+      if: ${{ github.repository != 'cyrusimap/cyrus-imapd' }}
+      shell: bash
+      run: |
+        git remote add upstream https://github.com/cyrusimap/cyrus-imapd.git
+        # n.b. --no-tags does not mean "no tags", it means "no automatic tag
+        # following".  we're explicitly fetching the tags we want, we do not
+        # need every other tag that's reachable from them
+        git fetch --no-tags upstream 'refs/tags/cyrus-imapd-*:refs/tags/cyrus-imapd-*'
+    - name: configure and build
+      shell: bash
+      run: |
+        echo "building cyrus version" $(./tools/git-version.sh)
+        ./tools/build-with-cyruslibs.sh
+    - name: make check with valgrind
+      shell: bash
+      run: |
+        make VG=1 check
```

Comment:
check discreet takes 18m. The regular check is much faster and the output is readable enough to find the real errors I think

Reply:
Yeah that seems fine.  In my mind, the main utility of running valgrind in CI is regularly knowing _if_ something is leaky.  Once you know something's leaky, you're probably going to do targeted local valgrind runs in order to track down and fix the leak, so the CI output doesn't really need to be diagnostically useful in its own right.